src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionFixProvider.cs

```diff
@@ -0,0 +1,169 @@
+            var methodOrProperty = throwNode.FirstAncestorOrSelf<MemberDeclarationSyntax>();
+            Contract.ThrowIfFalse(methodOrProperty is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax);
+
+            if (!memberReferencesBuilder.ContainsKey(methodOrProperty))
+            {
+                var memberSymbol = semanticModel.GetRequiredDeclaredSymbol(methodOrProperty, cancellationToken);
+                var references = await FindReferencesAsync(document, memberSymbol, cancellationToken).ConfigureAwait(false);
+                memberReferencesBuilder.Add(methodOrProperty, references);
+            }
+        }
+
+        var copilotService = document.GetRequiredLanguageService<ICopilotCodeAnalysisService>();
+        var memberImplementationDetails = await copilotService.ImplementNotImplementedExceptionsAsync(document, memberReferencesBuilder.ToImmutable(), cancellationToken).ConfigureAwait(false);
+
+        foreach (var syntaxNode in memberReferencesBuilder.Keys)
+        {
+            Contract.ThrowIfFalse(syntaxNode is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax);
+            var methodOrProperty = (MemberDeclarationSyntax)syntaxNode;
+
+            SyntaxNode? replacement;
+            if (!memberImplementationDetails.TryGetValue(methodOrProperty, out var implementationDetails))
+            {
+                replacement = AddErrorComment(methodOrProperty);
+                Logger.Log(FunctionId.Copilot_Implement_NotImplementedException_Failed, logLevel: LogLevel.Error);
+            }
+            else
+            {
+                replacement = implementationDetails.ReplacementNode;
+                if (replacement != null && (replacement is BasePropertyDeclarationSyntax or BaseMethodDeclarationSyntax))
+                {
+                    replacement = replacement
+                        .WithLeadingTrivia(methodOrProperty.GetLeadingTrivia())
+                        .WithTrailingTrivia(methodOrProperty.GetTrailingTrivia())
+                        .WithAdditionalAnnotations(Formatter.Annotation, WarningAnnotation, Simplifier.Annotation);
+                }
+                else
+                {
+                    replacement = AddErrorComment(methodOrProperty, implementationDetails.Message);
+                    Logger.Log(FunctionId.Copilot_Implement_NotImplementedException_Failed, logLevel: LogLevel.Error);
+                }
+            }
+
+            editor.ReplaceNode(methodOrProperty, replacement);
+        }
+
+        var changedRoot = editor.GetChangedRoot();
+        var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(cancellationToken).ConfigureAwait(false);
+        var spansToFormat = FormattingExtensions.GetAnnotatedSpans(changedRoot, new());
+        var formattedRoot = CSharpSyntaxFormatting.Instance.GetFormattingResult(changedRoot, spansToFormat, formattingOptions, new(), cancellationToken).GetFormattedRoot(cancellationToken);
```

Comment:
```suggestion
        var formattingOptions = await document.GetSyntaxFormattingOptionsAsync(cancellationToken).ConfigureAwait(false);
        var spansToFormat = FormattingExtensions.GetAnnotatedSpans(changedRoot, new());
        var formattedRoot = CSharpSyntaxFormatting.Instance.GetFormattingResult(changedRoot, spansToFormat, formattingOptions, new(), cancellationToken).GetFormattedRoot(cancellationToken);
```

why do you need any of this?